app/controllers/materials_controller.rb

```diff
@@ -0,0 +1,221 @@
+require 'rubygems'
```

Comment:
No one prevents you from creating a service: https://blog.engineyard.com/2014/keeping-your-rails-controllers-dry-with-services